tests/ssl-opt.sh

```diff
@@ -219,6 +230,9 @@ skip_next_test() {
 # Check if the required configuration ($1) is enabled
 is_config_enabled()
 {
+    if [ "$LIST_TESTS" -gt 0 ];then
```

Comment:
Why put this in every function? Wouldn't it be simpler to let the config checks happen, but then ignore `SKIP_NEXT` when listing test cases?

Reply:
The if conditions in every function was a hacky solution to a problem that I ran into when trying to just ignore the `SKIP_NEXT` option in `run_test()`:

In most of these functions, `P_QUERY` was being used and the associated program could have not been compiled at the time of calling.

To solve this, after your review, I defined `P_QUERY=":"` so that calling this function is ignored and the if conditions would always be false.

I was not able to solve this issue for the case of P_SRV without adding more complicated logic, so I just left it that way.
Hope this has been simplified now :)